--- smbwebclient.orig.php	Thu Apr  8 12:47:18 2004
+++ smbwebclient.php	Sat Apr 10 21:37:48 2004
@@ -1027,7 +1027,7 @@
   if ($items_are_files) {
     $html .=
     "<th width=\"20\" align=\"right\">".
-    "<input type=\"checkbox\" name=\"chkall\" onclick=\"javascript:sel_all()\" />".
+    "<input type=\"checkbox\" name=\"chkall\" onclick=\"javascript:sel_all(this)\" />".
     "</th>";
   }
   $html .= "<th width=\"20\" align=\"right\">".
@@ -1140,12 +1140,12 @@
   # do check all
   $this->javascript =
   "<script language=\"JavaScript\">\n".
-  "  function sel_all() {\n".
+  "  function sel_all(master_select) {\n".
   "    with (document.d_form) {\n".
   "      for (i=0; i<elements.length; i++) {\n".
   "        ele = elements[i];\n".
   "      if (ele.type==\"checkbox\")\n".
-  "          ele.checked = ! ele.checked;\n".
+  "          ele.checked = master_select.checked;\n".
   "      }\n".
   "    }\n".
   "  }\